File: 455A-Boredom.py
n = int(input())
arr = list(map(int, input().split()))

MAX = 100000
cnt = [0] * (MAX + 1)

mx = 0
for x in arr:
    cnt[x] += 1
    mx = max(mx, x)
dp = [0] * (mx + 1)

dp[0] = 0
if mx >= 1:
    dp[1] = cnt[1] * 1

for i in range(2, mx + 1):
    dp[i] = max(dp[i - 1], dp[i - 2] + cnt[i] * i)

print(dp[mx])

File: 455A-Boredom.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<long long> cnt(100001, 0);

    int x;
    int mx = 0;
    for (int i = 0; i < n; i++) {
        cin >> x;
        cnt[x]++;
        mx = max(mx, x);
    }

    vector<long long> dp(mx + 1, 0);

    dp[0] = 0;
    dp[1] = cnt[1] * 1;

    for (int i = 2; i <= mx; i++) {
        dp[i] = max(dp[i - 1], dp[i - 2] + cnt[i] * i);
    }

    cout << dp[mx] << "\n";
    return 0;
}
